Sources/HummingbirdWebSocket/WebSocketContext.swift

```diff
@@ -19,16 +19,16 @@ import NIOCore
 public protocol WebSocketContextProtocol: Sendable {
     var logger: Logger { get }
     var allocator: ByteBufferAllocator { get }
-    init(logger: Logger, allocator: ByteBufferAllocator)
+    init(channel: Channel, logger: Logger)
 }
 
 /// Default implementation of ``WebSocketContextProtocol``
 public struct WebSocketContext: WebSocketContextProtocol {
     public let logger: Logger
     public let allocator: ByteBufferAllocator
```

Comment:
I can imagine that the remote address is useful

Reply:
Likewise the original HTTPRequest could be useful in a `WebSocketContextProtocol`?

Reply:
This is basically a duplicate of the requirements from the RequestContext so including the request might be hard. You get the request in the `shouldUpgrade` call so things like `permessage-deflate` can be negotiated there